src/fping.c

```diff
@@ -252,6 +252,8 @@ int socket6 = -1;
 int hints_ai_family = AF_UNSPEC;
 #endif
 
+volatile int status_snapshot = 0;
```

Comment:
I think "volatile" is wrong here. It is a qualifier reserved for embedded programs where the variable's value can change without the compiler knowing (memory-mapped IO, etc.).

Reply:
Values changed in signal handlers also fall in that class - the variable may be changed outside of the normal flow of execution that the compiler expects.  Signal handlers are processed asynchronously from the main flow of execution.

Reply:
You are right, I learned something :) But then, shouldn't it be of type sig_atomic_t ?

Reply:
Makes sense to be cross-platform correct (I think sig_atomic_t == int o most platforms, but TIL something too :) )  I'll update the PR and add it to this commit.

Reply:
Ok updated to use `sig_atomic_t`.

Think I got the whole squash commit thing right...